src/pycatcher/webapp.py

```diff
@@ -0,0 +1,152 @@
+import pandas as pd
+from pathlib import Path
+from flask import (
+    Flask,
+    request,
+    render_template,
+    redirect,
+    url_for,
+    jsonify,
+    flash
+)
+from . import create_app
+from .catch import detect_outliers_iqr
+from typing import List, Dict, Any
+
+
+class FileValidator:
+    """Handles file validation logic."""
+
+    def __init__(self, allowed_extensions: List[str]):
+        self.allowed_extensions = allowed_extensions
+
+    def is_allowed_file(self, filename: str) -> bool:
+        """
+        Check if the given filename has an allowed extension.
+
+        Args:
+            filename (str): Name of the file to check
+
+        Returns:
+            bool: True if file extension is allowed, False otherwise
+        """
+        return ('.' in filename and
+                filename.rsplit('.', 1)[1].lower() in self.allowed_extensions)
+
+
+class OutlierAnalyzer:
```

Comment:
Main logic for the outlier detection goes here and the response page will be added here. Need to add more functionality by leveraging other functions.